src/gc/gcimpl.h

```diff
@@ -274,6 +274,13 @@ class GCHeap : public IGCHeapInternal
 
     virtual void DescrGenerationsToProfiler (gen_walk_fn fn, void *context);
 
+    virtual void SetDirty(void* address, size_t size);
+
+    virtual void* GetTable();
```

Comment:
I do not think there needs any methods on IGCHeap.

There should be a method on the GCToEEInterface that the GC calls when any of the write barrier parameters change, and the updated write barrier parameters should be passed as argument of this callback. The EE should update the write barrier with the passed parameters inside this callback - it won't need to call methods like `GetTable`  to get them.